package/MDAnalysis/core/universe.py

```diff
@@ -200,15 +204,19 @@ def __init__(self, *args, **kwargs):
                 self.filename = None
             else:
                 self.filename = args[0]
-
                 parser = get_parser_for(self.filename, format=topology_format)
                 try:
                     with parser(self.filename) as p:
                         self._topology = p.parse()
-                except IOError as err:
-                    raise IOError("Failed to load from the topology file {0}"
-                                  " with parser {1}.\n"
-                                  "Error: {2}".format(self.filename, parser, err))
+                except (IOError, OSError) as err:
+                    if err.errno is not None and errno.errorcode[err.errno] in ['ENOENT', 'EACCES']:
```

Comment:
Only with your mocked object. I get it with a real file. `anyopen` makes quite a lot of black magic.